src/util.h

```diff
@@ -127,4 +127,26 @@ bool safe_uint64_add(uint64_t* a, uint64_t b);
             fprintf(stdout, "%s:%d: " format "\n", FILENAME, __LINE__, __VA_ARGS__); \
     } while (0)
 
+/**
+ * Struct definining a rw buffer (buffer + length).
+ */
+typedef struct {
+    /** Data buffer. */
+    uint8_t* data;
```

Comment:
somewhere in here a `const` can be added to indicate that the data pointer can't be changed (but the contents can)

Reply:
Then one should just pass in a `const buffer_t`!

Reply:
*Admits defeat* f82a689